expr/types.go

```diff
@@ -80,6 +82,17 @@ type argString struct {
 func (a argString) Key() string    { return a.key }
 func (a argString) Optional() bool { return a.opt }
 
+// like string, but should result in a regex
+type argRegex struct {
+	key       string
+	opt       bool
+	validator []validator
+	val       **regexp.Regexp
```

Comment:
hmm, it is possible to do this without the double pointer, by initializing FuncAliasSub.Search to &regexp.Regexp{}, and then assinging *val = *re.Copy() 

but the double pointer appears to be the most straight forward approach